apis/logging/v1/input_source_types.go

```diff
@@ -0,0 +1,52 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NOTE: The Enum validation on SourceSpec.Type must be updated if the list of types changes.
```

Comment:
Don't we actually need type in the spec?  Will this break upgrades in future if we add others and existing deployments don't have "type" set?

Reply:
Since the spec is never empty, the type is redundant 
`{ http: { blah } }` vs. `{ type:http, http: {blah}}`
However if think we should  include a type for consistency with similar patterns elsewhere then I could be convinced.

Reply:
Maybe the better question is to ask if we ever intend to support a different input source (YAGNI?)  ?  We mostly are defining this input for internal usage and we are dictating the transport and protocol.  I don't honestly see why we need to consider supporting multiple and it benefits us to simply state: this is what it is and if you want to talk to us this is how you do it.